tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,247 @@
+generate animations using the :class:`~matplotlib.animation` module. An
+animation is a sequence of frames where each frame corresponds to a plot on a
+:class:`~matplotlib.figure.Figure`. This tutorial covers a general guideline on
+how to create such animations and the different options available.
+"""
+
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import numpy as np
+
+###############################################################################
+# Animation Classes
+# =================
+#
+# The process of animation can be thought about in 2 different ways:
+#
+# 1. Generate data for first frame and then modify this data for each frame to
+# create an animated plot.
+# 2. Generate a list (iterable) of plots (/images) that will each be a frame in
+# the animation.
+#
+# Both of these thought processes can be translated into matplotlib using the
+# :class:`~matplotlib.animation`'s :class:`~matplotlib.animation.FuncAnimation`
+# and :class:`~matplotlib.animation.ArtistAnimation` classes respectively.
+#
+# :class:`~matplotlib.animation.FuncAnimation`
+# --------------------------------------------
+#
+# :class:`~matplotlib.animation.FuncAnimation` class allows us to create an
+# animation by passing a function that iteratively modifies the data of a plot.
+# This is achieved by using the *setter* methods on various
+# :class:`~matplotlib.artist.Artist`
+# (examples: :class:`~matplotlib.lines.Line2D`,
+# :class:`~matplotlib.collections.PathCollection`, etc.). A usual
+# :class:`~matplotlib.animation.FuncAnimation` object takes a
+# :class:`~matplotlib.figure.Figure` that we want to animate and a function
+# *func* that modifies the data plotted on the figure. It uses the *frames*
+# parameter to determine the length of the animation. The *interval* parameter
+# is used to determine time in milliseconds between drawing of two frames.
+# (This is different than *fps* in `.animation.save` that we will
+# cover later). We will now look at some examples of using
+# :class:`~matplotlib.animation.FuncAnimation` with different artists.
+
+###############################################################################
+# Animating :class:`~matplotlib.lines.Line2D`
+# ---------------------------------------------------------
+#
+# `.pyplot.plot` returns a :class:`~matplotlib.lines.Line2D` collection. The
+# data on this collection can be modified by using the
+# `.lines.Line2D.set_data` function. Therefore we can use this to modify the
+# plot using the function for every frame.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+
+xdata, ydata = [], []
+line, = ax.plot(xdata, ydata, c='b')
+ax.grid()
+ax.set_ylim(-1, 1)
+
+
+def update(frame):
+    xdata.append(frame)
+    ydata.append(np.sin(2 * np.pi * frame / 30))
+    xmin, xmax = ax.get_xlim()
+
+    if frame >= xmax:
+        ax.set_xlim(xmin, 2*xmax)
+        ax.figure.canvas.draw()
```

Comment:
It seems to be working locally for me. Also works when I save the animation and check it. I think the reason why it isn't working in the example is because the xlim is already set to be quite big and it isn't getting a chance to extend before it repeats?

Reply:
I was just repurposing an existing example, but now that you mention it, I don't need that anymore, with the new example setup.